### DISCUSSION TITLE
blog/optimizely/optimizely-automatically-register-dependencies

### DISCUSSION THREAD

**Discussion by giscus** [dariusz-wozniak/dariuszwozniak.net-v2#7] [General]
# blog/optimizely/optimizely-automatically-register-dependencies

How to automatically register dependencies in Optimizely? (also in any other project that uses Microsoft.Extensions.DependencyInjection library)

https://dariuszwozniak.net/blog/optimizely/optimizely-automatically-register-dependencies

**Comment by bbsdddougla** [comment]
Cool tech. Any gotchas? Things to watch out for with Optimizely in particular? I noticed you exclude attributes. Anything else that shouldn't be registered by default?

**Reply by dariusz-wozniak** [reply]
I've used that convention-based automated registration in many production environments of non-Optimizely and Optimizely projects, and I haven't seen any trouble so far. The thing to remember is to have `RegistrationStrategy.Skip` flag and set all other registrations before the code. Also, you may add more assemblies in the `FromAssembliesOf` method.

For some Opti-infrastructure-related stuff (such as `IContentApiModelFilter`, `IMigrationStep`), I'm still using implicit registration via the `ServiceConfiguration`.
